File: src/gestortaller/Terrestre.java
package gestortaller;
import java.io.Serializable;

import gestorviajes.*;
public abstract class Terrestre implements Serializable{

	
	
	protected final  int GASTO_GASOLINA = 120;
	private String nombre;
	private Transportadora transportadora;
	private boolean descompuesto;
	private Silla[] SILLAS_ECONOMICAS;
	private Silla[] SILLAS_EJECUTIVAS;

	protected Terrestre(String nombre, Transportadora transportadora) {
		this.nombre = nombre;
		this.transportadora = transportadora;
	}

	public Transportadora getTransportadora() {
		return transportadora;
	}

	public void setTransportadora(Transportadora transportadora) {
		this.transportadora = transportadora;
	}

	public Silla[] getSILLASECONOMICAS() {
		return SILLAS_ECONOMICAS;
	}

	public void setSILLASECONOMICAS(Silla[] sILLAS_ECONOMICAS) {
		SILLAS_ECONOMICAS = sILLAS_ECONOMICAS;
	}

	public Silla[] getSILLASEJECUTIVAS() {
		return SILLAS_EJECUTIVAS;
	}

	public void setSILLASEJECUTIVAS(Silla[] sILLAS_EJECUTIVAS) {
		SILLAS_EJECUTIVAS = sILLAS_EJECUTIVAS;
	}

	public int getGastoGasolina() {
		return GASTO_GASOLINA;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}


	public boolean isDescompuesto() {
		return descompuesto;
	}

	public void setDescompuesto(boolean descompuesto) {
		this.descompuesto = descompuesto;
	}

	public String toString() {
		return this.nombre;
	}

	public Silla buscarSillaPorUbicacionyTipo(Ubicacion ubicacion, String tipo) {
		if (tipo.equalsIgnoreCase("ECONOMICA")) {
			for (Silla i : SILLAS_ECONOMICAS) {
				if (i.isEstado() & i.getUbicacion().equals(ubicacion)) {
					return i;
				}
			}
		} else if (tipo.equalsIgnoreCase("EJECUTIVA")) {
			for (Silla i : SILLAS_EJECUTIVAS) {
				if (i.isEstado() & i.getUbicacion().equals(ubicacion)) {
					return i;
				}
			}
		}
		return null;
	}
	
	public String Calcular_Sillas_Ocupadas() {
		int cont = 0;
		for (Silla i : this.getSILLASECONOMICAS()) {
			if (i.isEstado()) {
				cont += 1;
			}
		}
		for (Silla j : this.getSILLASEJECUTIVAS()) {
			if (j.isEstado()) {
				cont += 1;
			}
		}
		return "Esta es la cantidad de silla ocupadas:"+cont;
	}

	public abstract double Calcular_Consumo_Gasolina(double distancia_en_km);
}

File: practica02/Practica2VALE/Practica02VALE/SRC/gestorAplicacion/parada/Terrestre.py

class Terrestre:
   

 
    def __init__(self, nombre, transportadora):
       
        self.Gasto_gasolina = 120
        self._descompuesto = False
        self._SILLAS_ECONOMICAS = []
        self._SILLAS_EJECUTIVAS = []
        self._nombre = nombre
        self._transportadora= transportadora

    
    def getTransportadora(self):
        return self._transportadora

    def setTransportadora(self, transportadora):
        self._transportadora = transportadora

    def getSILLASECONOMICAS(self):
        return self._SILLAS_ECONOMICAS

    def setSILLASECONOMICAS(self, sILLAS_ECONOMICAS):
        self._SILLAS_ECONOMICAS = sILLAS_ECONOMICAS

    def getSILLASEJECUTIVAS(self):
        return self._SILLAS_EJECUTIVAS

    def setSILLASEJECUTIVAS(self, sILLAS_EJECUTIVAS):
        self._SILLAS_EJECUTIVAS = sILLAS_EJECUTIVAS

    def getGastoGasolina(self):
        return self.Gasto_gasolina

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre



    def isDescompuesto(self):
        return self._descompuesto

    def setDescompuesto(self, descompuesto):
        self._descompuesto = descompuesto

    def __str__(self):
        return self._nombre


    def buscarSillaPorUbicacionyTipo(self, ubicacion, tipo):

        if tipo.lower() == "ECONOMICA".lower():
            for i in self._SILLAS_ECONOMICAS:
                if i.isEstado() and i.getUbicacion() == ubicacion:
                    return i
        elif tipo.lower() == "EJECUTIVA".lower():
            for i in self._SILLAS_EJECUTIVAS:
                if i.isEstado() and i.getUbicacion() == ubicacion:
                    return i
        return None
  
    
    def Calcular_Sillas_Ocupadas(self):
        pass

#gasolina
    def Calcular_Consumo_Gasolina(self, distancia_en_km):
        pass
